From 8ee22276173095092864d1558f017caeb544f168 Mon Sep 17 00:00:00 2001
From: Othmane AHL ZOUAOUI <othmane.ahlzouaoui@st.com>
Date: Wed, 31 Jul 2024 11:10:22 +0200
Subject: [PATCH 6/8] onnxruntime: mlas: fix -mcpu incompatibility flag

The "-mcpu=" switch is introduced externaly into CMake by Yocto
buildsystem toolchain. This switch is not compatible with MLAS
build mechanism and causes the its compilation break due to
"unsupported instructions".
This switch needs to be removed for MLAS in order to isolate
the changes only for this submodule and its depencencies.

Signed-off-by: Othmane AHL ZOUAOUI <othmane.ahlzouaoui@st.com>
---
 cmake/onnxruntime_mlas.cmake | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/cmake/onnxruntime_mlas.cmake b/cmake/onnxruntime_mlas.cmake
index 682dcfc5fe..98803f9dcf 100644
--- a/cmake/onnxruntime_mlas.cmake
+++ b/cmake/onnxruntime_mlas.cmake
@@ -354,6 +354,17 @@ else()
         set_source_files_properties(${MLAS_SRC_DIR}/sqnbitgemm_kernel_neon.cpp
                                     PROPERTIES COMPILE_FLAGS " -march=armv8.2-a+dotprod")
         if (NOT APPLE)
+          string(REGEX REPLACE "-mcpu=[-a-zA-Z0-9_.^$*+?]*" "" _tmp ${CMAKE_C_FLAGS})
+          set(CMAKE_C_FLAGS ${_tmp})
+          message("CMAKE_C_FLAGS: ${CMAKE_C_FLAGS}")
+
+          string(REGEX REPLACE "-mcpu=[-a-zA-Z0-9_.^$*+?]*" "" _tmp ${CMAKE_ASM_FLAGS})
+          set(CMAKE_ASM_FLAGS ${_tmp})
+          message("CMAKE_ASM_FLAGS: ${CMAKE_ASM_FLAGS}")
+
+          string(REGEX REPLACE "-mcpu=[-a-zA-Z0-9_.^$*+?]*" "" _tmp ${CMAKE_CXX_FLAGS})
+          set(CMAKE_CXX_FLAGS ${_tmp})
+          message("CMAKE_CXX_FLAGS: ${CMAKE_CXX_FLAGS}")
           set(mlas_platform_srcs
             ${mlas_platform_srcs}
             ${MLAS_SRC_DIR}/aarch64/HalfGemmKernelNeon.S
--
2.34.1
